File: ios-samples/poslink/EposPosLinkExample/EposPosLinkExample/Transactions/TransactionRecord.swift
import Foundation
import TeyaUnifiedEposSDK

struct TransactionRecord: Identifiable, Equatable {
    let id: String
    let type: TeyaTransactionType
    let isSuccess: Bool
    let amountMinor: Int
    let currency: String
    let gatewayPaymentId: String?
    let timestamp: Int64
    var isRefunded: Bool = false

    var isRefundable: Bool {
        type == .payment && isSuccess && !isRefunded && gatewayPaymentId != nil
    }
}

File: java-samples/poslink/EposPosLinkExample/src/main/java/com/example/eposposlinkexample/models/TransactionRecord.java
package com.example.eposposlinkexample.models;

import com.teya.unifiedepossdk.models.TransactionType;

public record TransactionRecord(String id, TransactionType type, boolean successful, int amountMinor, String currency,
                                String gatewayPaymentId, long timestamp, boolean refunded, boolean cashPayment,
                                boolean tabPayment) {

    public boolean refundable() {
        return type == TransactionType.Payment && successful && !refunded && gatewayPaymentId != null && !cashPayment;
    }

    public TransactionRecord asRefunded() {
        return new TransactionRecord(id, type, successful, amountMinor, currency, gatewayPaymentId, timestamp,
                true, cashPayment, tabPayment);
    }
}
